packages/nimble-components/src/table/tests/table.stories.ts

```diff
@@ -48,10 +49,19 @@ const dataSetIdFieldNames = {
 
 const overviewText = 'The `nimble-table` is a component that offers a way to render tabular data in a variety of ways in each column.';
 
-const dataDescription = `\`data\` is a property that is an array of records. A record provides the data that backs a single row in the table.
-Each record is made up of fields, which are key/value pairs. The key in each pair must be of type \`string\`, which is defined by the type
-\`TableFieldName\`. The value in each pair must be of type \`string\`, \`number\`, \`boolean\`, \`Date\`, \`null\`, or \`undefined\`,
-which is defined by the type \`TableFieldValue\`.`;
+const dataDescription = `To set the data on the table, call \`setData()\` with an array data records. Each record is made up of fields,
+which are key/value pairs. The key in each pair must be of type \`string\`, which is defined by the type \`TableFieldName\`. The value
+in each pair must be of type \`string\`, \`number\`, \`boolean\`, \`Date\`, \`null\`, or \`undefined\`, which is defined by the type \`TableFieldValue\`.
+
```

Comment:
I'd vote to be really explicit about when it's necessary to call `setData()` if a client wants to update the table. Something like this maybe (assuming my understanding is correct)?

```suggestion
The table will not automatically update if the contents of the array change after calling `setData()`. To trigger an update you can call `setData()` again with the same array reference.
```